### PR TITLE
Add Scala greeter example

### PR DESCRIPTION
**Pull request by gvdongen**
- Adds a Scala keyed greeter examples

### CODE REVIEW

File: scala/hello-world-http/build.sbt [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import sbt.Keys._
+import sbt._
+
+name := "hello-world-http"
+version := "1.0"
+scalaVersion := "2.13.12"
```
Comment (slinkydeveloper) [inline scala/hello-world-http/build.sbt:6]: isn't scala 2.13 legacy? shouldn't we provide an example for scala 3?

**Review by slinkydeveloper — CHANGES_REQUESTED** [review]

File: scala/hello-world-http/build.sbt (hunk 1 above)
Comment (gvdongen) [reply scala/hello-world-http/build.sbt:6]: Isn't scala 2.13 still used more? I can switch though. Probably it matter little here, if all dependencies support Scala 3

Comment (slinkydeveloper) [reply scala/hello-world-http/build.sbt:6]: IDK the data about usage of scala 2.13 but I would rather prefer to show an example with the latest Scala version. If someone is stuck on 2.13, they'll probably find a way to make it working.

Comment (gvdongen) [reply scala/hello-world-http/build.sbt:6]: I'll merge this one and try adding a Scala 3 example as well

Comment (tillrohrmann) [reply scala/hello-world-http/build.sbt:6]: > I'll merge this one and try adding a Scala 3 example as well

+1 for this approach.

File: scala/hello-world-http/src/main/scala/dev/restate/sdk/examples/Greeter.scala [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package dev.restate.sdk.examples
+
+import dev.restate.sdk.RestateService
+import dev.restate.sdk.common.{CoreSerdes, StateKey}
+import greeter.GreeterGrpc.GreeterImplBase
+import greeter.{GreetRequest, GreetResponse}
+import io.grpc.stub.StreamObserver
+
+class Greeter extends GreeterImplBase with RestateService {
+  private val COUNT = StateKey.of[Integer]("count", CoreSerdes.INT)
```
Comment (tillrohrmann) [inline scala/hello-world-http/src/main/scala/dev/restate/sdk/examples/Greeter.scala:10]: Don't statics go into a companion object usually in Scala?

File: scala/hello-world-http/src/main/scala/dev/restate/sdk/examples/Greeter.scala [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+  override def greet(request: GreetRequest, responseObserver: StreamObserver[GreetResponse]): Unit = {
+    val ctx = restateContext
+
+    val count = ctx.get(COUNT).orElse(Integer.valueOf(1))
+    ctx.set(COUNT, Integer.valueOf(count + 1))
+
+    responseObserver.onNext(GreetResponse.newBuilder.setMessage("Hello " + request.getName + " for the " + count + " time!").build)
```
Comment (tillrohrmann) [inline scala/hello-world-http/src/main/scala/dev/restate/sdk/examples/Greeter.scala:18]:
```suggestion
    responseObserver.onNext(GreetResponse.newBuilder.setMessage(s"Hello ${request.getName} for the ${count} time!").build)
```

**Review by tillrohrmann — APPROVED** [review]
Thanks a lot for creating this PR @gvdongen. This is really good for helping users who are interested in Scala and SBT :-) +1 for merging and tackling Scala 3 as a potential follow-up.

**Pull request merged**